libraries/AP_Networking/AP_Networking_tests.cpp

```diff
@@ -200,4 +205,72 @@ void AP_Networking::test_TCP_reflect(void)
     }
 }
 
+void AP_Networking::test_connector_loopback(void)
+{
+    startup_wait();
+    GCS_SEND_TEXT(MAV_SEVERITY_INFO, "Connector Loopback: starting");
+
+    // start tcp discard server
+    auto *listen_sock = new SocketAPM(false);
+    if (listen_sock == nullptr) {
+        GCS_SEND_TEXT(MAV_SEVERITY_ERROR, "connector_loopback: failed to create socket");
+        return;
+    }
+    listen_sock->reuseaddress();
+
+    // use netif ip address
+    char ipstr[16];
+    SocketAPM::inet_addr_to_str(get_ip_active(), ipstr, sizeof(ipstr));
+    if (!listen_sock->bind(ipstr, 9)) {
+        GCS_SEND_TEXT(MAV_SEVERITY_ERROR, "connector_loopback: failed to bind");
+        return;
+    }
+    GCS_SEND_TEXT(MAV_SEVERITY_INFO, "connector_loopback: bound");
+
+    if (!listen_sock->listen(1)) {
+        GCS_SEND_TEXT(MAV_SEVERITY_ERROR, "connector_loopback: failed to listen");
+        return;
+    }
+    GCS_SEND_TEXT(MAV_SEVERITY_INFO, "connector_loopback: listening");
+    // create discard client
+    auto *client = new SocketAPM(false);
+    if (client == nullptr) {
+        GCS_SEND_TEXT(MAV_SEVERITY_ERROR, "connector_loopback: failed to create client");
+        return;
+    }
+    while (!client->connect(ipstr, 9)) {
+        hal.scheduler->delay(10);
+    }
+    GCS_SEND_TEXT(MAV_SEVERITY_INFO, "connector_loopback: connected");
+
+    // accept client
+    auto *sock = listen_sock->accept(100);
+    if (sock == nullptr) {
+        GCS_SEND_TEXT(MAV_SEVERITY_ERROR, "connector_loopback: failed to accept");
+        return;
+    }
+
+    uint8_t buf[1024];
+    uint32_t last_report_ms = AP_HAL::millis();
+    uint32_t total_rx = 0;
+    while (true) {
+        if ((param.tests & TEST_CONNECTOR_LOOPBACK) == 0) {
+            hal.scheduler->delay(1);
+            continue;
+        }
+        client->send(buf, 1024);
```

Comment:
I'm pretty sure Valgrind would not like this; maybe just zero it on the stack.

I'd actually suggest something non-zero in case there's some ridiculous optimisation somewhere.  Perhaps:
```
for (uint8_t i=0; i<ARRAY_SIZE(buf); i++) {
    buf[i] = i;
}
```

Reply:
changed to using memset to set all the values to 0xab